deform/linear_deformation.py

```diff
@@ -0,0 +1,1173 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Example of shape-based image reconstruction using linearized deformations.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+from builtins import super
+from numbers import Number
+from odl.operator.operator import Operator, OperatorComp
+import odl
+import numpy as np
+import time
+import matplotlib.pyplot as plt
+standard_library.install_aliases()
+
+
+class Functional(Operator):
```

Comment:
If the code can not possibly be written without this, this should be a merge into the functionals branch.

With that said, I'm 99% certain we do not need functionals for this to work, so the dependency should be removed.